integrationtests/test_all_examples.py

```diff
@@ -45,133 +144,60 @@
     TUTORIALS,
 ]
 
-
-#: Mapping from filepath to a callable() -> (skipped: bool, reason: str)
-FILES_MAY_BE_SKIPPED = {}
-
-
-def skip_file_if(condition, reason, filepath):
-    """ Mark a file to be skipped for a given condition.
-
-    Parameters
-    ----------
-    condition: callable() -> bool
-        The condition for skipping a file.
-    reason : str
-        Reason for skipping the file.
-    filepath : str
-        Path of the file which may be skipped from tests.
-    """
-    def wrapped():
-        return condition(), reason
-
-    filepath = os.path.abspath(filepath)
-    FILES_MAY_BE_SKIPPED[filepath] = wrapped
-
-
-skip_file_if(
+SEARCHER = ExampleSearcher(source_dirs=SOURCE_DIRS)
+SEARCHER.skip_file_if(
     lambda: True, "Not a target file to be tested",
     os.path.join(DEMO, "demo.py"),
 )
-skip_file_if(
+SEARCHER.skip_file_if(
```

Comment:
Nit: Naturally I would expect to read "'file' is skipped if 'condition' for 'reason'", so making `file_path` the first argument would be more readable.

Reply:
Yep. Done.